### PR TITLE
[UIE-102] Fail tests that write console logs

### PR DESCRIPTION
**Pull request by nawatts**
https://broadworkbench.atlassian.net/browse/UIE-102

The unit tests tend to grow noisier over time with warnings and other console logs until there's enough noise that it becomes difficult to find informative output to debug test failures.

I just wrapped up a round of cleanup: https://github.com/DataBiosphere/terra-ui/pulls?q=is%3Apr+UIE-97

To avoid this problem in the future, this change fails any test that logs to the console.

### CODE REVIEW

**Review by aherbst-broad — APPROVED** [review]

**Review by aherbst-broad — APPROVED** [review]

**Review by s-rubenstein — COMMENTED** [review]
Presumably this will also fail tests locally, not just on circle? Could we make it fail on circleci specifically? And if not, can we document how to turn it off for people who want to use console logging for local development? (I assume it’s just flip these to false, but since it could be hard to find this file, maybe throw in a line pointing here in the readme?

**Comment by nawatts** [comment]
I'd prefer to keep the behavior the same locally and in CI to avoid the situation where you believe your tests are passing from running them locally and then find out they fail only after pushing to GitHub and waiting for CI. I think this tightens the feedback loop.

**Comment by nawatts** [comment]
@s-rubenstein I added an escape hatch in https://github.com/DataBiosphere/terra-ui/pull/4064/commits/7c1a62f2266940d9d86568e97a52ef24d9dc0baf. Setting an `ALLOW_LOGS` environment variable allows using logs in tests without failing the tests. Is that acceptable?

**Review by pshapiro4broad — APPROVED** [review]
🎉

File: src/setupTests.ts [hunk 1]
```diff
@@ -2,6 +2,21 @@ import '@testing-library/jest-dom';
 import 'blob-polyfill';
 
 import { toHaveNoViolations } from 'jest-axe';
+import failOnConsole from 'jest-fail-on-console';
+
+// Fail tests that produce console logs.
+// Console warnings or errors suggest there are issues with the test.
+// Other console logs are noise that make it harder to find informative output when tests do fail.
+if (!process.env.ALLOW_LOGS) {
```
Comment (cahrens) [inline src/setupTests.ts:10]: Please make sure to add this to the wiki.

**Review by cahrens — APPROVED** [review]
Thanks @nawatts !

**Review by s-rubenstein — APPROVED** [review]
Looks good! Thanks for adding the escape hatch!

File: src/setupTests.ts (hunk 1 above)
Comment (nawatts) [reply src/setupTests.ts:10]: Added to https://github.com/DataBiosphere/terra-ui/wiki/Getting-started#testing.

**Pull request merged**